File: python/largest_series_product.py
from functools import reduce


def largest_product(series, size):
    if series == '':
        raise ValueError('span must be smaller than string length')

    if size < 0:
        raise ValueError('span must not be negative')

    if size > len(series):
        raise ValueError('span must be smaller than string length')

    try:
        lst = [int(c) for c in series]
    except Exception as e:
        raise ValueError('digits input must only contain digits')

    ans = 0
    for i in range(len(lst) - size + 1):
        ans = max(ans, reduce(lambda a, b: a * b, lst[i:i + size]))

    return ans

File: dart/largest_series_product.dart
import 'dart:math';

class LargestSeriesProduct {
  int largestProduct(input, span) {
    if (span < 0) {
      throw ArgumentError('span must not be negative');
    }
    if (input.length < span) {
      throw ArgumentError('span must be smaller than string length');
    }

    final List<int> lst = [];
    for (int i = 0; i < input.length; i++) {
      final n = int.tryParse(input[i]);
      if (n == null) {
        throw ArgumentError('digits input must only contain digits');
      }
      lst.add(n);
    }

    int res = 0;
    for (int i = 0; i < lst.length - span + 1; i += 1) {
      int r = 1;
      for (int j = 0; j < span; j++) {
        r *= lst[i + j];
      }
      res = max(res, r);
    }
    return res;
  }
}
